### ISSUE TITLE
save debug symbols before stripping

### ISSUE THREAD

**Issue by r10s** [deltachat/deltachat-android#1437]
for getting a grip on crashes that are reported from time to time, it would be useful to have the debug-symbols that are stripped from the release-version somewhere.

(having an extra "fat" version that includes the debug symbols seems to be of less use as this would require users to reproduce a crash - which is not always easily doable. also it may require users installing an extra version, maybe going the export/import way which may cause additional problems, esp. when having few disk space)

as @link2xt pointed out on irc, the native part of Delta Chat android, `libnative-utils.so`,  is an ELF:

```
> ELF 64-bit LSB shared object, ARM aarch64, version 1 (SYSV), dynamically linked, 
> BuildID[sha1]=ede5bfc42dc08bbf929672906d737f49371b86c9, stripped
> this is what "file" says
> I have unpacked it from the APK
> so all I need is debug symbols for this ELF stored separately until someone reports crash 
> or just an unstripped binary, that is equivalent to having a mapfile
> so before you strip the binary, save a copy somewhere 
```

the "stripping" is done implicitly in https://github.com/deltachat/deltachat-android/blob/master/jni/Android.mk#L20 (or at least, for me it is hidden :) - so, any hint/pr to achieve that would be very useful :)

**Comment by link2xt** [comment]
Found this: https://stackoverflow.com/questions/37575043/not-strip-debug-symbols-ndk-build

If this does not increase the size of binary too much, maybe it's even possible to keep the symbols in the binary to get proper backtraces. Otherwise, save .so somewhere and strip the binary afterwards with `strip`.

**Comment by link2xt** [comment]
Also this thread: https://groups.google.com/forum/#!topic/android-ndk/Q7PlBW2dmBk

**Comment by r10s** [comment]
also came over that:
https://developer.android.com/ndk/guides/ndk-stack.html
(via https://stackoverflow.com/questions/40355200/how-to-avoid-stripping-for-native-code-symbols-for-android-app/40368127)

`ndk-stack` is a tool that proposes to replace the hex-numbers in an android-log by symbol-names and linenumbers - sounds nice. it needs the log and the raw .a object files from eg. obj/local/armeabi-v7a - so, these have to be saved, in my normal environment, they are just overwritten with the next build.

i am also checking  for the release 1.10.3 how much larger an .apk with debug symbols in would be:
- 55647703 bytes for gplay/release/deltachat-gplay-release-1.10.3-stripped-as-usual.apk. 
- ...

**Comment by link2xt** [comment]
Related: #1134

**Comment by r10s** [comment]
closed by #1443
